### PR TITLE
feat(widget): control custom recipient

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

New `disableCustomRecipient` option in the injected widget.
The option hides the custom recipient toggle from swap/limit/twap widgets settings.
It also resets the custom recipient (if it's set) when true.

# To Test

1. Open the widget configurator and paste `{"disableCustomRecipient": true}` in "Raw JSON parameters". Press "Update widget"
2. Open each swap/limit/twap widgets -> open settings
- [ ] there should not be "Custom recipient" control displayed

# Testing demo

https://github.com/user-attachments/assets/39d53bd1-4b8a-4e27-b968-7491aacd663b

## Summary by CodeRabbit

* **New Features**
  * New widget setting to disable the custom recipient option across trading widgets.
  * When enabled, the custom recipient control and its toggle are hidden and the recipient is auto-reset.
  * Recipient controls and toggle behavior are now consistent across swap, limit, TWAP and yield widgets; cross-chain/non-EVM flows still require a recipient and keep the toggle enabled (but may be non-interactive).

### CODE REVIEW

**Review by fairlighteth — COMMENTED** [review]
<details>
  <summary>✅ <strong>Manual verification:</strong> widget configurator flow works</summary>

  - I manually tested this in the widget configurator.
  - Result: the configured widget updated correctly and the flow worked as expected.
  - Side note: should we add `disableCustomRecipient` as a configurable option in the widget configurator side panel?

  </details>

  <details>
  <summary>⚠️ <strong>AI Review</strong> (Codex GPT-5, worked 5m): 2 actionable findings</summary>

  **Finding 1: `disableCustomRecipient` can make non-EVM bridge swaps impossible**

  - Location: `apps/cowswap-frontend/src/modules/trade/hooks/useWithRecipient.ts:25`
  - With `disableCustomRecipient: true`, `useIsWithRecipient` returns `false` before the non-EVM bridge branch can show the required recipient input. `useResetRecipient` also clears the recipient.
  - Bridging still remains enabled unless `disableCrossChainSwap` is separately set, and validation then raises `RecipientNotSet` for non-EVM output chains. The user gets a disabled "Recipient is required for ..." action with no input available to fix it.

  **Suggested fix**
  - Either keep `SetRecipient` visible for required non-EVM bridge receivers even when optional custom recipients are disabled, or have this option also disable the affected bridge path.
  - Add a focused test for `disableCustomRecipient=true`, non-EVM bridging, and no recipient.

  **Finding 2: new trade hooks bypass the frontend module import convention**

  - Location: `apps/cowswap-frontend/src/modules/trade/hooks/useResetRecipient.ts:14` and `apps/cowswap-frontend/src/modules/trade/hooks/useWithRecipient.ts:5`
  - These are module-to-module imports from `modules/trade` to `modules/injectedWidget`, but `apps/cowswap-frontend/AGENTS.md` requires cross-module imports to go through the root alias barrel.
  - The same issue was already fixed in `SwapWidget`, but these two added imports still use `../../injectedWidget`.

  **Suggested fix**
  - Replace both imports with `import { useInjectedWidgetParams } from 'modules/injectedWidget'`.

  <details>
  <summary>🤖 Prompt for AI agents</summary>

  ```text
  Verify these findings against current code. Fix only still-valid issues, keep changes minimal, and validate with targeted tests.

  Context:
  - For disableCustomRecipient=true plus non-EVM bridging, useIsWithRecipient hides SetRecipient and useResetRecipient clears recipient, while validateTradeForm still requires a recipient for non-EVM output chains.
  - In useResetRecipient.ts and useWithRecipient.ts, replace cross-module relative imports from ../../injectedWidget with the modules/injectedWidget barrel import.
  </details>
  </details>
  ```

**Comment by shoom3301** [comment]
> Finding 1

This is expected and added to the option docs: `If you want always having recipient === order owner, then set disableCrossChainSwap=true as well`

>Finding 2

Thanks! FIxed

**Review by fairlighteth — COMMENTED** [review]
<details>
  <summary>⚠️ <strong>AI Review</strong> (Codex GPT-5, worked 4m): follow-up still reproducible</summary>

  **Rechecked**
  - Author response: @shoom3301 said Finding 1 is expected and documented via `disableCrossChainSwap=true`; Finding 2 was fixed.
  - Code path: `apps/cowswap-frontend/src/modules/trade/hooks/useWithRecipient.ts:25`, `apps/cowswap-frontend/src/modules/trade/hooks/useResetRecipient.ts:51`, `apps/cowswap-frontend/src/modules/swap/containers/SwapWidget/index.tsx:214`, `apps/cowswap-frontend/src/
  modules/tradeFormValidation/services/validateTradeForm.ts:168`.
  - Test coverage: existing recipient visibility tests do not cover `disableCustomRecipient=true` with non-EVM bridging.

  **Result:** Still broken. Documenting “also set `disableCrossChainSwap=true`” is not enough because the widget currently allows an invalid runtime combination: `disableCustomRecipient=true` plus non-EVM cross-chain swaps enabled. In that state, the required recipient
  input is hidden and cleared while validation still requires a recipient, leaving the user stuck on a disabled “Recipient is required...” action.

  **Suggested next step**
  - Either make `disableCustomRecipient` also disable the affected non-EVM bridge path, or keep the required non-EVM recipient input visible while hiding only optional custom-recipient controls.
  - Add a focused test for `disableCustomRecipient=true`, non-EVM bridging, and empty recipient.

  <details>
  <summary>Review scope and related context</summary>

  - Finding 2 is addressed: `useResetRecipient.ts` and `useWithRecipient.ts` now import `useInjectedWidgetParams` from `modules/injectedWidget`.
  - This follow-up is limited to the prior Finding 1 runtime behavior. The docs now mention pairing with `disableCrossChainSwap`, but current code does not enforce that pairing.

  </details>

  <details>
  <summary>🤖 Prompt for AI agents</summary>

  ```text
  Verify this follow-up finding against current code. Fix only if still valid, keep the change minimal, and validate with targeted tests.

  Context:
  - Prior Finding 1: disableCustomRecipient=true plus non-EVM bridging hides SetRecipient and clears recipient, while validation still requires recipient.
  - Current code still returns false in useIsWithRecipient before the non-EVM bridging branch and clears recipient when disableCustomRecipient is true.
  - Bridging remains enabled unless disableCrossChainSwap is set.
  - Expected fix: either keep required non-EVM recipient input visible or make disableCustomRecipient also disable the affected bridge path. Add focused test coverage.
  ```

  </details>

  </details>

**Comment by shoom3301** [comment]
>AI Review (Codex GPT-5, worked 4m): follow-up still reproducible

I still think this is ok. It's the integrator's responsibility to read the option description.

**Review by fairlighteth — APPROVED** [review]
Approving. Trusting your judgement on finding 2 👍🏼

File: libs/widget-lib/src/types.ts [hunk 1]
```diff
@@ -371,6 +371,13 @@ export interface CowSwapWidgetParams {
    */
   disableCrossChainSwap?: boolean
 
+  /**
+   * Disabled setting custom recipient for all trading widgets (swap,limit,twap)
```
Comment (kernelwhisperer) [inline libs/widget-lib/src/types.ts]: Is this a typo?

```diff
- Disabled
+ Disable
```

**Review by kernelwhisperer — APPROVED** [review]
LGTM
